ui/app/routes/vault/cluster/access/method.ts

```diff
@@ -0,0 +1,42 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+
+import Route from '@ember/routing/route';
+import { service } from '@ember/service';
+import { supportedManagedAuthBackends } from 'vault/helpers/supported-managed-auth-backends';
+import AuthMethodResource from 'vault/resources/auth/method';
+
+import type ApiService from 'vault/services/api';
+import type PathHelpService from 'vault/services/path-help';
+
+export default class VaultClusterAccessMethodRoute extends Route {
+  @service declare readonly api: ApiService;
+  @service declare readonly pathHelp: PathHelpService;
+
+  async model(params: { path: string }) {
+    const { path } = params;
+    const { auth } = await this.api.sys.internalUiListEnabledVisibleMounts();
+    const methods = this.api
+      .responseObjectToArray(auth, 'path')
+      .map((method) => new AuthMethodResource(method, this));
+    const method = methods.find((m) => m.id === path);
+    // the user could have entered a random path in the URL that doesn't correspond to an existing method
+    if (method) {
+      const supportManaged = supportedManagedAuthBackends();
+      // do not fetch path-help for unmanaged auth types
+      if (!supportManaged.includes(method.methodType)) {
+        method.paths = { apiPath: method.apiPath, paths: [], itemTypes: [] };
+        return method;
+      }
+      return this.pathHelp.getPaths(method.apiPath, path, '', '').then((pathInfo) => {
+        method.paths = pathInfo;
+        return method;
+      });
+    } else {
+      // throw a 404 if the path doesn't match any of the fetched methods
+      throw { httpStatus: 404, path };
```

Comment:
While using the API service I actually had a question about error handling. The old application adapter used `httpStatus` but the new api service uses `status` - should we be consistent about which key houses the HTTP error code? Errors thrown by the new api service don't render in the [error template](https://github.com/hashicorp/vault/blob/main/ui/app/templates/error.hbs#L0-L1) the same because of this difference which got me thinking we should probably make them the same?

Reply:
I actually made a ticket to standarize the error handling since there are numerous different templates that are slightly different but I haven't got around to it yet. Early on I updated the [Page::Error](https://github.com/hashicorp/vault/blob/main/ui/lib/core/addon/components/page/error.ts) component to handle both Ember Data and API Client errors and was thinking we should just use that component everywhere.

Reply:
Awesome! That sounds great - glad you've already thought of it!